tests/pull.bats

```diff
@@ -312,3 +312,18 @@ load helpers
 
   run_buildah rmi alpine
 }
+
+@test "pull image with TMPDIR set" {
+  testdir=${TESTDIR}/buildah-test
+  mkdir -p $testdir
+  mount -t tmpfs -o size=1M tmpfs $testdir
+  export TMPDIR=$testdir
+
+  run_buildah 125 pull --signature-policy ${TESTSDIR}/policy.json alpine
```

Comment:
May I recommend collapsing those two into one?
```bash
TMPDIR=$testdir run_buildah ....
```
As it is now, if the `expect_output` fails, `$TMPDIR` will remain set for all subsequent tests, and very likely cause cascading failures.

Even with this, a failure will leave `$testdir` mounted. That is much harder to catch, so I'm going to sweep it under the carpet.